core/common/src/main/java/alluxio/collections/IndexedSet.java

```
@@ -255,46 +231,40 @@ public void remove() {
+    return index != null && index.contains(value);
   }
 
   /**
    * Gets a subset of objects with the specified field value. If there is no object with the
-   * specified field value, a newly created empty set is returned. Otherwise, the returned set is
-   * backed up by an internal set, so changes in internal set will be reflected in returned set.
+   * specified field value, a newly created empty set is returned.
    *
-   * @param index the field index
+   * @param indexDefinition the field index definition
    * @param value the field value to be satisfied
    * @return the set of objects or an empty set if no such object exists
    */
-  // TODO(gpang): Remove this method, if it is not being used.
-  public Set<T> getByField(FieldIndex<T> index, Object value) {
-    Set<T> set = getByFieldInternal(index, value);
-    return set == null ? new HashSet<T>() : Collections.unmodifiableSet(set);
+  public Set<T> getByField(IndexDefinition<T> indexDefinition, Object value) {
+    FieldIndex<T> index = mIndices.get(indexDefinition);
+    return index == null ? Collections.<T>emptySet() : index.getByField(value);
   }
 
   /**
-   * Gets the first object from the set of objects with the specified field value.
+   * Gets the object from the set of objects with the specified non-unique field value.
    *
-   * @param index the field index
+   * @param indexDefinition the field index definition
    * @param value the field value
    * @return the object or null if there is no such object
    */
-  public T getFirstByField(FieldIndex<T> index, Object value) {
-    Set<T> all = getByFieldInternal(index, value);
-    try {
-      return all == null || !all.iterator().hasNext() ? null : all.iterator().next();
-    } catch (NoSuchElementException e) {
-      return null;
-    }
+  public T getFirstByField(IndexDefinition<T> indexDefinition, Object value) {
+    FieldIndex<T> index = mIndices.get(indexDefinition);
+    return index == null ? null : index.getFirst(value);
```

this should throw an exception if index is null